docs/shared/recipes/running-tasks/customizing-inputs.md

```diff
@@ -44,136 +48,282 @@ To understand some of the examples below, let's imagine the following simple wor
 
 {% /graph %}
 
-## Global vs per-project inputs
+## Types of Inputs
 
-Tasks can have `inputs` defined for them [globally in the `nx.json` file](/reference/nx-json#inputs-&-namedinputs) or [on a per-project basis in the project configuration](/reference/project-configuration#inputs-&-namedinputs).
+Nx can consider the following types of Inputs when computing the hash for a given operation.
 
-{% tabs %}
-{% tab label="Global" %}
+### Project Configuration
 
-```json {% fileName="nx.json" highlightLines=[4] %}
-{
-  "targetDefaults": {
-    "build": {
-      "inputs": ["..."]
-    }
+The configuration of a project includes the definition of the task itself. Changing this will often result in a change of behavior of running different tasks.
+
+Nx always considers the configuration of the project of a task and its dependencies when calculating the hash of a task.
+
+### Command Arguments
+
+Passing different arguments to a Nx command will often result in a change of behavior of running tasks. Nx will filter out arguments which are for Nx itself and do not have any effect on running of tasks such as `--parallel` or `--projects`. Nx will always take into consideration command arguments when calculating the hash for a task.
```

Comment:
```suggestion
Passing different arguments to a Nx command will often result in a task changing behavior. Because of this, Nx will take into consideration command arguments when calculating the hash for a task. Nx will filter out arguments which are for Nx itself and do not have any effect on the actual tasks such as `--parallel` or `--projects`.
```